File: vue/apps/web-antd/src/api/monitor/cache.ts
/**
 * @description 缓存监控相关接口
 * @Link  https://github.com/xiujiecn/xiu-admin
 * @Copyright  Copyright (c) 2025 XiuAdmin CLI
 * @Author  Lxj <li@xiujie.cn>
 * @License  https://github.com/xiujiecn/xiu-admin/blob/master/LICENSE
 * @date 2024-03-21
 */

import { requestClient } from '#/api/request';


export interface CommandStats {
  name: string;
  value: string;
}

export interface RedisInfo {
  [key: string]: string;
}

export interface CacheInfo {
  commandStats: CommandStats[];
  dbSize: number;
  info: RedisInfo;
}

export async function redisCacheInfo() {
  return requestClient.get<CacheInfo>('/monitor/cache');
}
  

File: server/api/monitor/v1/cache.go
// Package v1
// @Link  https://github.com/xiujiecn/xiu-admin
// @Copyright  Copyright (c) 2025	 LiXiujie
// @Author  Lxj <li@xiujie.cn>
// @License  https://github.com/xiujiecn/xiu-admin/blob/master/LICENSE
package v1

import (
	"github.com/gogf/gf/v2/frame/g"
)

type RedisInfoReq struct {
	g.Meta `path:"/cache" method:"get" tags:"监控-缓存监控" summary:"获取缓存监控信息" x-check-permission:"cpm:monitor:cache:list"`
}

type RedisInfoRes struct {
	DBSize       int               `json:"dbSize"`
	CommandStats []*CommandStats   `json:"commandStats"`
	Info         map[string]string `json:"info"`
}

type CommandStats struct {
	Name  string `json:"name"`
	Value string `json:"value"`
}
